lib/internal/make-api-call.js

```diff
@@ -75,12 +75,17 @@ exports.inject = function(options) {
     var useClientId = query.supportsClientId && clientId && clientSecret;
     delete query.supportsClientId;
 
-    var requestUrl = formatRequestUrl(path, query, useClientId);
+    var method = query.method;
+    delete query.method;
+
+    var isPost = method == 'POST'
+    var postData = isPost ? query : null;
+    var requestUrl = formatRequestUrl(path, query, method, useClientId);
 
     function rateLimitedGet() {
       return requestQueue.add(function() {
         return Task.start(function(resolve, reject) {
-          return makeUrlRequest(requestUrl, resolve, reject);
+          return makeUrlRequest(requestUrl, resolve, reject, postData);
```

Comment:
Sure, but the current APIs and HTTP methods are a known quantity, and if they change we would need to add support for them explicitly in the client library, so I think it's OK for the client library to only deal with the known set of HTTP methods until they change (which may never happen).